wordpress-mobile/WordPress-Android
# Media upload error codes in editor

This PR is part of a series related to #12325 and should improve the error codes presented to the user in the GB Editor when media upload errors occur.

I finally decided to leverage the `EditorFragmentListener` interface we already use adding the `onFailedToInsertMediaTapped` method.

From UI side I tried to stay near to what we have in iOS where we show this dialog when tapping on a failed to upload media

<p align="center">
  <img width="300" src=https://user-images.githubusercontent.com/47797566/90237062-f8589d00-de23-11ea-964b-a60ff50adc2c.png>
</p>

And added the error message to the dialog we already have in GB

| Overlay on upload fail | On tap Dialog Before | On tap Dialog After |
|---|---|---|
| ![image](https://user-images.githubusercontent.com/47797566/90240079-e2011000-de28-11ea-8658-f1cd917bddc3.png) | ![image](https://user-images.githubusercontent.com/47797566/90238253-ebd54400-de25-11ea-8e3b-a33b10d7def9.png) | ![image](https://user-images.githubusercontent.com/47797566/90238225-de1fbe80-de25-11ea-8348-5bf2b787541c.png) | 

Maybe we can arrange the message in a different way like we could just use the Title field of the dialog, populating it with the error message if we have one or using the generic `Retry uploading?` string if we do not have a valid one. But I would need your thoughts on the above @osullivanchris, thanks 🙇‍♂️.

### To test
This must be tested both with WP.com/JP and oure self-hosted sites. You should get the same behavior/error messages that you get in #12620 and #12625 but using the editor as per below steps (notice that you should be able to trigger the upload errors as per steps in #12620 and #12625)

- Create a new post in GB 
- add an image/video/gallery (in this last case with multiple images)
- try to upload image(s)/video
- trigger the error so that you get a failed upload media
- tap on the failed to upload media and notice the dialog with the error message (note the error message is the same you should get when uploading the media from the Media Library)
- smoke test the dialog action buttons
- Close the editor, come again to the Draft post and tap on the failed to upload media. Notice you still get the same error message and you are able to use the dialog action buttons.

PR submission checklist:
